From 19c9a74ae80074714cc5251ea58cbda284f8061f Mon Sep 17 00:00:00 2001
From: Austen Dicken <cvpcsm@gmail.com>
Date: Wed, 23 Nov 2011 01:03:46 -0600
Subject: [PATCH] build static e2fsprogs

change makefies to build static mke2fs and tune2fs and place them in
$(PRODUCT_OUT)/utilities

Change-Id: I0139b3dcfe64a4031db7719ddd71120d174d1990

static e2fsck, tune2fs, and mke2fs for recovery

Change-Id: I8725cc296be8967265261e92da743c149e9ec45b

resize: allow static build

Change-Id: Id5dabaf87de775bb7461f83a9e7ae9ca754a1635

e2fsprogs: Clean up makefiles

 * Remove duplicate modules

Change-Id: If68c5685a76943bc6bfdcb630c15c2f9ac59effd
---
 e2fsck/Android.mk    | 42 ++++++++++++++++++++++++++++++++++
 lib/quota/Android.mk |  3 ++-
 lib/uuid/Android.mk  | 12 ++++++++++
 misc/Android.mk      | 64 ++++++++++++++++++++++++++++++++++++++++++++++++++++
 resize/Android.mk    | 13 +++++++++++
 5 files changed, 133 insertions(+), 1 deletion(-)

diff --git a/e2fsck/Android.mk b/e2fsck/Android.mk
index 0789298..b54d665 100644
--- a/e2fsck/Android.mk
+++ b/e2fsck/Android.mk
@@ -56,6 +56,18 @@ include $(BUILD_SHARED_LIBRARY)
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(libext2_profile_src_files)
+LOCAL_STATIC_LIBRARIES := $(libext2_profile_system_shared_libraries) $(libext2_profile_shared_libraries)
+LOCAL_C_INCLUDES := $(libext2_profile_c_includes)
+LOCAL_CFLAGS := $(libext2_profile_cflags)
+LOCAL_PRELINK_MODULE := false
+LOCAL_MODULE := libext2_profile
+LOCAL_MODULE_TAGS := optional
+
+include $(BUILD_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(libext2_profile_src_files)
 LOCAL_SHARED_LIBRARIES := $(addsuffix -host, $(libext2_profile_shared_libraries))
 LOCAL_C_INCLUDES := $(libext2_profile_c_includes)
 LOCAL_CFLAGS := $(libext2_profile_cflags)
@@ -156,6 +168,36 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(e2fsck_src_files)
 LOCAL_C_INCLUDES := $(e2fsck_c_includes)
 LOCAL_CFLAGS := $(e2fsck_cflags)
+LOCAL_STATIC_LIBRARIES := $(e2fsck_system_shared_libraries) $(e2fsck_shared_libraries) libext2fs
+LOCAL_MODULE := recovery_e2fsck
+LOCAL_MODULE_CLASS := RECOVERY_EXECUTABLES
+LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
+LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/recovery
+LOCAL_MODULE_STEM := e2fsck
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+LOCAL_MODULE_TAGS := optional
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(e2fsck_src_files)
+LOCAL_C_INCLUDES := $(e2fsck_c_includes)
+LOCAL_CFLAGS := $(e2fsck_cflags)
+LOCAL_STATIC_LIBRARIES := $(e2fsck_system_shared_libraries) $(e2fsck_shared_libraries) libext2fs
+LOCAL_MODULE := utility_e2fsck
+LOCAL_MODULE_CLASS := UTILITY_EXECUTABLES
+LOCAL_MODULE_PATH := $(PRODUCT_OUT)/utilities
+LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/utilities
+LOCAL_MODULE_STEM := e2fsck
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+LOCAL_MODULE_TAGS := optional
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(e2fsck_src_files)
+LOCAL_C_INCLUDES := $(e2fsck_c_includes)
+LOCAL_CFLAGS := $(e2fsck_cflags)
 LOCAL_SHARED_LIBRARIES := $(addsuffix -host, $(e2fsck_shared_libraries))
 LOCAL_MODULE := e2fsck_host
 LOCAL_MODULE_STEM := e2fsck
diff --git a/lib/quota/Android.mk b/lib/quota/Android.mk
index d102ec3..facd9cc 100644
--- a/lib/quota/Android.mk
+++ b/lib/quota/Android.mk
@@ -55,7 +55,8 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(libext2_quota_src_files)
 LOCAL_C_INCLUDES := $(libext2_quota_c_includes)
 LOCAL_CFLAGS := $(libext2_quota_cflags)
-LOCAL_STATIC_LIBRARIES := libc $(libext2_quota_static_libraries)
+LOCAL_SYSTEM_STATIC_LIBRARIES := libc $(libext2_quota_shared_libraries) libext2fs
+LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE := libext2_quota
 LOCAL_MODULE_TAGS := optional
 
diff --git a/lib/uuid/Android.mk b/lib/uuid/Android.mk
index b26521e..94e65aa 100644
--- a/lib/uuid/Android.mk
+++ b/lib/uuid/Android.mk
@@ -62,6 +62,18 @@ LOCAL_SRC_FILES := $(libext2_uuid_src_files)
 LOCAL_C_INCLUDES := $(libext2_uuid_c_includes)
 LOCAL_CFLAGS := $(libext2_uuid_cflags)
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
+LOCAL_STATIC_LIBRARIES := $(libext2_uuid_system_shared_libraries)
+LOCAL_MODULE := libext2_uuid
+LOCAL_MODULE_TAGS := eng
+LOCAL_PRELINK_MODULE := false
+
+include $(BUILD_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(libext2_uuid_src_files)
+LOCAL_C_INCLUDES := $(libext2_uuid_c_includes)
+LOCAL_CFLAGS := $(libext2_uuid_cflags)
 LOCAL_MODULE := libext2_uuid-host
 LOCAL_MODULE_TAGS := optional
 
diff --git a/misc/Android.mk b/misc/Android.mk
index 79ba61d..1dab374 100644
--- a/misc/Android.mk
+++ b/misc/Android.mk
@@ -68,6 +68,38 @@ include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(mke2fs_src_files)
 LOCAL_C_INCLUDES := $(mke2fs_c_includes)
+LOCAL_CFLAGS := $(mke2fs_cflags) $(mke2fs_cflags_linux)
+LOCAL_STATIC_LIBRARIES := $(mke2fs_shared_libraries)
+LOCAL_STATIC_LIBRARIES += $(mke2fs_system_shared_libraries) libext2fs
+LOCAL_MODULE := recovery_mke2fs
+LOCAL_MODULE_TAGS := eng
+LOCAL_MODULE_CLASS := RECOVERY_EXECUTABLES
+LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
+LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/recovery
+LOCAL_MODULE_STEM := mke2fs
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(mke2fs_src_files)
+LOCAL_C_INCLUDES := $(mke2fs_c_includes)
+LOCAL_CFLAGS := $(mke2fs_cflags) $(mke2fs_cflags_linux)
+LOCAL_STATIC_LIBRARIES := $(mke2fs_shared_libraries)
+LOCAL_STATIC_LIBRARIES += $(mke2fs_system_shared_libraries) libext2fs
+LOCAL_MODULE := utility_mke2fs
+LOCAL_MODULE_TAGS := eng
+LOCAL_MODULE_CLASS := UTILITY_EXECUTABLES
+LOCAL_MODULE_PATH := $(PRODUCT_OUT)/utilities
+LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/utilities
+LOCAL_MODULE_STEM := mke2fs
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(mke2fs_src_files)
+LOCAL_C_INCLUDES := $(mke2fs_c_includes)
 ifeq ($(HOST_OS),linux)
 LOCAL_CFLAGS := $(mke2fs_cflags) $(mke2fs_cflags_linux)
 else
@@ -181,6 +213,38 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(tune2fs_src_files)
 LOCAL_C_INCLUDES := $(tune2fs_c_includes)
 LOCAL_CFLAGS := $(tune2fs_cflags)
+LOCAL_STATIC_LIBRARIES := $(tune2fs_shared_libraries) $(tune2fs_system_shared_libraries) libext2fs
+LOCAL_MODULE := utility_tune2fs
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := UTILITY_EXECUTABLES
+LOCAL_MODULE_PATH := $(PRODUCT_OUT)/utilities
+LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/utilities
+LOCAL_MODULE_STEM := tune2fs
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(tune2fs_src_files)
+LOCAL_C_INCLUDES := $(tune2fs_c_includes)
+LOCAL_CFLAGS := $(tune2fs_cflags)
+LOCAL_STATIC_LIBRARIES := $(tune2fs_shared_libraries) $(tune2fs_system_shared_libraries) libext2fs
+LOCAL_MODULE := recovery_tune2fs
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := RECOVERY_EXECUTABLES
+LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
+LOCAL_UNSTRIPPED_PATH := $(PRODUCT_OUT)/symbols/recovery
+LOCAL_MODULE_STEM := tune2fs
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(tune2fs_src_files)
+LOCAL_C_INCLUDES := $(tune2fs_c_includes)
+LOCAL_CFLAGS := $(tune2fs_cflags)
 LOCAL_SHARED_LIBRARIES := $(addsuffix -host, $(tune2fs_shared_libraries))
 LOCAL_MODULE := tune2fs_host
 LOCAL_MODULE_STEM := tune2fs
diff --git a/resize/Android.mk b/resize/Android.mk
index 42b25ce..047859e 100644
--- a/resize/Android.mk
+++ b/resize/Android.mk
@@ -61,6 +61,19 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(resize2fs_src_files)
 LOCAL_C_INCLUDES := $(resize2fs_c_includes)
 LOCAL_CFLAGS := $(resize2fs_cflags)
+LOCAL_STATIC_LIBRARIES := $(resize2fs_shared_libraries)
+LOCAL_STATIC_LIBRARIES += $(resize2fs_system_shared_libraries)
+LOCAL_MODULE := resize2fs_static
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_PATH := $(PRODUCT_OUT)/install/bin
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := $(resize2fs_src_files)
+LOCAL_C_INCLUDES := $(resize2fs_c_includes)
+LOCAL_CFLAGS := $(resize2fs_cflags)
 LOCAL_SHARED_LIBRARIES := $(addsuffix -host, $(resize2fs_shared_libraries))
 LOCAL_MODULE := resize2fs_host
 LOCAL_MODULE_STEM := resize2fs
-- 
2.10.0

